File: 238-product-of-array-except-self/product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res=[]
        #left to right
        x = 1
        for i in nums:
            res.append(x)
            x = x *i
        x = 1
        for i in range(len(nums)-1,-1,-1):
            res[i] *= x
            x = x*nums[i]
        return res

File: 238-product-of-array-except-self/product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> res;
        int x = 1;
        for(int i=0;i<nums.size();i++) {
            res.push_back(x);
            x = x*nums[i];
        }
        x = 1;
        for(int i=nums.size() -1;i>=0;i--) {
            res[i] *= x;
            x = x*nums[i];
        }
        return res;
    }
};